src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtils.java

```diff
@@ -732,16 +756,37 @@ private static void fixOctal(@NonNull Pod podFromYaml) {
     }
 
     private static void convertDecimalIntegersToOctal(List<KeyToPath> items) {
-        items.stream().forEach(i -> {
-            Integer mode = i.getMode();
+        items.forEach(i -> {
+            var mode = i.getMode();
             if (mode != null) {
-                i.setMode(convertToOctal(mode));
+                i.setMode(convertPermissionToOctal(mode));
             }
         });
     }
 
-    private static int convertToOctal(Integer defaultMode) {
-        return Integer.parseInt(Integer.toString(defaultMode, 10), 8);
+    /**
+     * Permissions are generally expressed in octal notation, e.g. 0777.
+     * After parsing, this is stored as the integer 777, but the snakeyaml-engine does not convert to decimal first.
+     * When the client later sends the pod spec to the server, it sends the integer as is through the json schema,
+     * however the server expects a decimal, which means an integer between 0 and 511.
+     *
+     * The user can also provide permissions as a decimal integer, e.g. 511.
+     *
+     * This method attempts to guess whether the user provided a decimal or octal integer, and converts to octal if needed,
```

Comment:
I wonder if we could make this more reliable by checking the octal representation ([reference](https://en.wikipedia.org/wiki/File-system_permissions#Numeric_notation)). Actual modes will normally have the group bits match either the user bits or the other bits (it would be weird to have three different octets). And normally u ≥ g ≥ o and w ≥ r ≥ x right? For example `777` satisfies all constraints whereas `511` in octal would be `-r-x--x--x` which violates r ≥ x for g/o and is thus technically legal for exotic scenarios but unlikely to be what the user meant.